Express server crashes when there is an error

Whenever I get an error 500 or 404 in the backend, the entire Express server crashes. I think this needs to be fixed.

```Finding user with ID: undefined
Class {
  error: 'NOT_FOUND',
  message: 'Could not find what you are looking for',
  statusCode: 404 }

/Applications/MAMP/htdocs/neo-project-ullrich/routes/user.js:10
            throw err;
            ^
Could not find what you are looking for(NOT_FOUND)[Http code 404]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! neo-project-ullrich@0.0.0 start: `ng build && node ./bin/www`
npm ERR! Exit status 1```

Would it be possible to redirect to some page instead? This way we don't have to restart the (local) server and re-authenticate.

--- a/routes/team.js
+++ b/routes/team.js
@@ -10,7 +10,7 @@ router.get('/:id', function(req, res, next) {
         if(err){
             //send back error response
             next(err);
-        } else{
+        } else {
             res.status(200).send(record);
         }
         
@@ -29,7 +29,7 @@ router.post('/', function(req, res){
         if(err){
             //send back error response
             next(err);
-        }else{
+        } else {
             res.status(200).send(record);
         }
         
@@ -41,72 +41,65 @@ router.get('/', function(req, res, next) {
         if(err){
             //send back error response
             next(err);
-        } else{
+        } else {
             res.status(200).send(record);
         }
-        
     })
 });
 
 //Team Entries
-router.get('/:id/entry', function(req, res) {
+router.get('/:id/entry', function(req, res, next) {
     let teamId = req.params.id;
     EntryTable.findByTeam(teamId, function(err, records){
         if(err){
-            console.log(`Error is ${err}`)
-            console.error(err);
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(records);
         }
-        res.status(200).send(records);
     })
 });
 
-router.post('/:id/entry', function(req, res){
+router.post('/:id/entry', function(req, res, next){
     EntryTable.createEntry(req.params.id, req.body, function(err, record){
         if(err){
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(record);
         }
-        res.status(200).send(record);
     })
 });
 
 //Entry Drivers
-router.get('/:teamId/entry/:entryId/drivers', function(req, res) {
+router.get('/:teamId/entry/:entryId/drivers', function(req, res, next) {
     let entryId = req.params.entryId;
     DriverTable.findByEntry(entryId, function(err, records){
         if(err){
-            console.log(`Error is ${err}`)
-            console.error(err);
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(records);
         }
-        res.status(200).send(records);
     })
 });
 
-router.post('/:teamId/entry/:entryId/drivers', function(req, res){
+router.post('/:teamId/entry/:entryId/drivers', function(req, res, next){
     DriverTable.createEntry(req.params.entryId, req.body, function(err, record){
         if(err){
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(record);
         }
-        res.status(200).send(record);
     })
 });
 
 //Team Drivers
-router.get('/:teamId/drivers', function(req, res) {
+router.get('/:teamId/drivers', function(req, res, next) {
     let teamId = req.params.teamId;
     DriverTable.findByTeam(teamId, function(err, records){
         if(err){
-            console.log(`Error is ${err}`)
-            console.error(err);
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(records);
         }
-        res.status(200).send(records);
     })
 });
 

--- a/routes/user.js
+++ b/routes/user.js
@@ -2,24 +2,24 @@ var express = require('express');
 var router = express.Router();
 var UserTable = require('../database/user');
 
-router.get('/:id', function(req, res) {
+router.get('/:id', function(req, res, next) {
     let userId = req.params.id;
     UserTable.findById(userId, function(err, record){
         if(err){
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(record);
         }
-        res.status(200).send(record);
     })
 });
 
-router.get('/', function(req, res) {
+router.get('/', function(req, res, next) {
     UserTable.getAllUsers(function(err, record){
         if(err){
-            //send back error response
-            throw err;
+            next(err);
+        } else {
+            res.status(200).send(record);
         }
-        res.status(200).send(record);
     })
 });
 

